src/main/java/nl/mrwouter/minetopiafarms/events/NPCClickListener.java

```diff
@@ -22,97 +20,46 @@ public void onNPCRightClick(NPCRightClickEvent e) {
-						if (clicker.getInventory().removeItem(item).size() == 0)
-							paymentAmount = paymentAmount
-									+ (Main.getPlugin().getConfig().getDouble("TerugverkoopPrijs.Mijnwerker.GOLD_ORE")
-											* item.getAmount());
-					} else if (item.getType() == XMaterial.LAPIS_ORE.parseMaterial()) {
-						if (clicker.getInventory().removeItem(item).size() == 0)
-							paymentAmount = paymentAmount
-									+ (Main.getPlugin().getConfig().getDouble("TerugverkoopPrijs.Mijnwerker.LAPIS_ORE")
-											* item.getAmount());
-					} else if (item.getType() == XMaterial.REDSTONE_ORE.parseMaterial()) {
-						if (clicker.getInventory().removeItem(item).size() == 0)
-							paymentAmount = paymentAmount + (Main.getPlugin().getConfig()
-									.getDouble("TerugverkoopPrijs.Mijnwerker.REDSTONE_ORE") * item.getAmount());
-					} else if (item.getType() == XMaterial.DIAMOND_ORE.parseMaterial()) {
-						if (clicker.getInventory().removeItem(item).size() == 0)
-							paymentAmount = paymentAmount + (Main.getPlugin().getConfig()
-									.getDouble("TerugverkoopPrijs.Mijnwerker.DIAMOND_ORE") * item.getAmount());
-					}
-				}
+			if(paymentAmount != 0){
+				API.getEcon().depositPlayer(clicker, paymentAmount);
+				clicker.sendMessage(
+						Main.getMessage("GeldBetaald").replaceAll("<Bedrag>", Utils.formatMoney(paymentAmount)));
+				API.updateScoreboard(clicker);
 			}
+		}
+	}
 
-			for (ItemStack item : clicker.getInventory().getContents()) {
-				if (item != null && item.getType() != null) {
-					if (item.getType().toString().contains("LOG")) {
-						if (clicker.getInventory().removeItem(item).size() == 0)
-							paymentAmount = paymentAmount + (item.getAmount()
-									* Main.getPlugin().getConfig().getInt("TerugverkoopPrijs.Houthakker"));
-					}
-				}
-			}
+	private double calculateItemPrice(ItemStack item, Player clicker){;
+		String itemName = XMaterial.matchXMaterial(item).name();
 
-			API.getEcon().depositPlayer(clicker, paymentAmount);
-			clicker.sendMessage(
-					Main.getMessage("GeldBetaald").replaceAll("<Bedrag>", Utils.formatMoney(paymentAmount)));
-			API.updateScoreboard(clicker);
+		Double configPrice = getJobItemPrice("Boer", itemName);
 
+		if (configPrice.doubleValue() == 0) {
+			configPrice = getJobItemPrice("Mijnwerker", itemName);
+
+			if (configPrice.doubleValue() == 0 &&
+					item.getType().toString().contains("LOG") &&
+					clicker.getInventory().removeItem(item).size() == 0)
+				return (item.getAmount() * Main.getPlugin().getConfig().getInt("TerugverkoopPrijs.Houthakker"));
 		}
+
+		if (configPrice.doubleValue() == 0 || XMaterial.matchXMaterial(item).parseItem() == null)
+			return 0;
+
+		if (clicker.getInventory().removeItem(item).size() == 0)
+			return configPrice.doubleValue() * item.getAmount();
+
+		return 0;
+	}
+
+	private double getJobItemPrice(String job, String itemName){
+		return Double.valueOf(Main.getPlugin().getConfig().getDouble("TerugverkoopPrijs." + job + "." + itemName));
```

Comment:
Onnodige `Double#valueOf(String)`, of mis ik iets?

Reply:
dat zie je goed